src/libraries/System.Console/src/System/ConsolePal.iOS.cs

```diff
@@ -0,0 +1,201 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Text;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    internal class NSLogStream : Stream
+    {
+        private bool _error;
+
+        public NSLogStream(bool error)
+        {
+            _error = error;
+        }
+
+        public override void Flush() { }
+
+        public override int Read(byte[] buffer, int offset, int count) => throw new PlatformNotSupportedException();
```

Comment:
Nit: the read/seek-related methods should just throw NotSupported rather than PlatformNotSupported; this is a write-only stream, so they'd they're regardless of platform